mobile/src/components/interview/components/history/history.tsx

```diff
@@ -1,9 +1,146 @@
 import React, { FC } from 'react';
 
-import { Text } from '~/components/common/common';
+import { ButtonVariant, DataStatus, PermissionKey } from '~/common/enums/enums';
+import { InterviewNoteCreateRequestDto } from '~/common/types/types';
+import {
+  Button,
+  Input,
+  ScrollView,
+  Spinner,
+  Stack,
+  Text,
+  View,
+} from '~/components/common/common';
+import { checkHasPermission } from '~/helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFocusEffect,
+  useState,
+} from '~/hooks/hooks';
+import { interviewActions } from '~/store/actions';
+import { interviewNotesCreateArguments } from '~/validation-schemas/validation-schemas';
+
+import { DEFAULT_CREATE_NOTE_PAYLOAD } from './common';
+import { NoteCardsList } from './components/components';
+import { styles } from './styles';
 
 const History: FC = () => {
-  return <Text>History</Text>;
+  const [isOpen, setIsOpen] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const { user, interviewId, interviewDataStatus, notes } = useAppSelector(
+    ({ auth, interview }) => ({
+      user: auth.user,
+      notes: interview.notes,
+      interviewId: interview.interview?.id,
+      interviewDataStatus: interview.interviewDataStatus,
+    }),
+  );
+
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<InterviewNoteCreateRequestDto>({
+      defaultValues: DEFAULT_CREATE_NOTE_PAYLOAD,
+      validationSchema: interviewNotesCreateArguments,
+    });
+
+  const isLoading = interviewDataStatus === DataStatus.PENDING;
+
+  const toggleInput = (): void => {
+    setIsOpen((prev) => !prev);
+    reset();
+  };
+
+  const hasPermission = checkHasPermission({
+    permissionKeys: [
+      PermissionKey.MANAGE_INTERVIEWS,
+      PermissionKey.MANAGE_INTERVIEW,
+    ],
+    userPermissions: user?.permissions ?? [],
+    checkMode: 'every',
+  });
+
+  const handleAdd = (payload: InterviewNoteCreateRequestDto): void => {
+    if (interviewId) {
+      const { note } = payload;
+      dispatch(interviewActions.createNote({ note, interviewId: interviewId }));
+      toggleInput();
+    }
+  };
+
+  useEffect(() => {
+    if (interviewId) {
+      dispatch(interviewActions.getNotes({ interviewId: Number(interviewId) }));
+    }
+  }, [interviewId]);
+
+  useFocusEffect(
+    useCallback(() => {
+      return () => {
+        setIsOpen(false);
+        reset();
+      };
+    }, []),
+  );
+
+  if (isLoading) {
+    return <Spinner isOverflow />;
+  }
```

Comment:
Can't we avoid spinner there and just disable buttons?

Reply:
Maybe pass isLoading to the button ?
<img src="https://user-images.githubusercontent.com/82529236/188300778-a65cb3c3-4cb2-46ad-9b64-9db734a4e28b.gif" width="40%">

Reply:
Yes, I meant this